src/java.base/share/classes/java/lang/runtime/SwitchBootstraps.java

```diff
@@ -52,15 +57,27 @@ private SwitchBootstraps() {}
 
     private static final MethodHandles.Lookup LOOKUP = MethodHandles.lookup();
 
-    private static final MethodHandle DO_TYPE_SWITCH;
-    private static final MethodHandle DO_ENUM_SWITCH;
+    private static final MethodHandle INSTANCEOF_CHECK;
+    private static final MethodHandle INTEGER_EQ_CHECK;
+    private static final MethodHandle OBJECT_EQ_CHECK;
+    private static final MethodHandle NULL_CHECK;
+    private static final MethodHandle IS_ZERO;
+    private static final MethodHandle ENUM_LOOKUP;
 
     static {
         try {
-            DO_TYPE_SWITCH = LOOKUP.findStatic(SwitchBootstraps.class, "doTypeSwitch",
-                                           MethodType.methodType(int.class, Object.class, int.class, Object[].class));
-            DO_ENUM_SWITCH = LOOKUP.findStatic(SwitchBootstraps.class, "doEnumSwitch",
-                                           MethodType.methodType(int.class, Enum.class, int.class, Object[].class));
+            INSTANCEOF_CHECK = LOOKUP.findStatic(SwitchBootstraps.class, "instanceofCheck",
+                                           MethodType.methodType(boolean.class, Object.class, Class.class));
+            INTEGER_EQ_CHECK = LOOKUP.findStatic(SwitchBootstraps.class, "integerEqCheck",
+                                           MethodType.methodType(boolean.class, Object.class, Integer.class));
+            OBJECT_EQ_CHECK = LOOKUP.findStatic(Objects.class, "equals",
+                                           MethodType.methodType(boolean.class, Object.class, Object.class));
+            NULL_CHECK = LOOKUP.findStatic(SwitchBootstraps.class, "nullCheck",
```

Comment:
```suggestion
            NULL_CHECK = LOOKUP.findStatic(Objects.class, "isNull",
```
Can remove the `nullCheck` method.